contracts/CompoundMiddleContract.sol

```diff
@@ -168,6 +168,75 @@ contract CompoundMiddleContract {
+        address _comptrollerAddress,
+        uint256 _amountToBorrowInWei,
+        address payable _leverageContractAddress
+    ) external returns (uint256) {
+        // declare references to external contracts
+        CEth cEth = CEth(_cEtherAddress);
+        Comptroller comptroller = Comptroller(_comptrollerAddress);
+        CErc20 cToken = CErc20(_cEtherAddress);
+        ComptrollerStatus memory getAccountLiquidityResponse = ComptrollerStatus(0, 0, 0);
+
+        require(cToken.balanceOf(address(this)) > 0, "DEPOSIT TOKENS FIRST");
+
+        // enter the market with the cTokens received (to make the above supplied tokens collateral)
+        address[] memory cTokens = new address[](1); // 1 is the array length
+        // cTokens is the list of tokens for which the market is to be entered
+        cTokens[0] = _cEtherAddress;
+        uint256[] memory errors = comptroller.enterMarkets(cTokens);
+        require(errors[0] == 0, "Comptroller.enterMarkets FAILED");
+
+        (getAccountLiquidityResponse.error2, getAccountLiquidityResponse.liquidity, getAccountLiquidityResponse.shortfall) = comptroller.getAccountLiquidity(address(this));
+        require(getAccountLiquidityResponse.error2 == 0, "comptroller.getAccountLiquidity FAILED");
+        require(getAccountLiquidityResponse.shortfall == 0, "account underwater");
+        require(getAccountLiquidityResponse.liquidity > 0, "account has excess collateral");
+
+        // liquidity is the USD value borrowable by the user, before it reaches liquidation
+        console.log("Liquidity available: ", getAccountLiquidityResponse.liquidity); 
+
+        // get current price of ETH from price feed
+        UniswapAnchoredView uniView = UniswapAnchoredView(0x046728da7cb8272284238bD3e47909823d63A58D);
+        // uint ethPrice = ;
+        // console.log("Eth price: ", ethPrice);
+        // liquidity is the maximum amount that can be borrowed in usd
+        // convert _amountToBorrowInWei to (scaled) usd
+        // console.log("amountToBorrowInWei: ", _amountToBorrowInWei);
+        // uint256 amountToBorrow = _amountToBorrowInWei * uniView.price("ETH");
+        // console.log("amountToBorrow: ", amountToBorrow/10**6);
+
+        // scale liquidity up to maintain precision
+        uint256 scaledLiquidity = getAccountLiquidityResponse.liquidity * (10**18);
+
+        // CHECK: IF USER IS ALLOWED TO BORROW THE AMOUNT ENTERED
+        require(_amountToBorrowInWei * uniView.price("ETH") <= scaledLiquidity, "BORROW FAILED: NOT ENOUGH COLLATERAL");
+
+        // borrow
+        // console.log(amountToBorrow, scaledLiquidity);
+        
+        require(cEth.borrow(_amountToBorrowInWei) == 0, "BORROW FAILED");
+
+        console.log("Borrowed %s wei", _amountToBorrowInWei);
+
+        uint256 borrowBalance = cEth.borrowBalanceCurrent(address(this));
+
+        // send ether to Leverage contract
+        (bool success, ) = _leverageContractAddress.call{ value: address(this).balance }("");
```

Comment:
Why this?
You can import the contract and inherit this function and call this function directly?